### DISCUSSION TITLE
Targeting .NET 8

### DISCUSSION THREAD

**Discussion by anna-is-cute** [goatcorp/Dalamud#1729] [Q&A] [answered]
I would have asked this question in the Discord, but you can't even get casual roles during the patch. Very helpful.

According to [the news site](https://dalamud.dev/news/2024/03/11/net8-upgrade/), the .NET 8 upgrade "should be mostly transparent to plugin devs", so I take that as meaning no API level bump and likely nothing broken outside of the noted breaking changes.

What I would like to know is if there is any need to update plugins specifically to target .NET 8. If not, is it all good to just leave them until the next natural update needed?

**Comment by Caraxi** [comment, marked as answer]
Dalamud will load NET7 plugins still, and as far as I'm aware theres no plan for API level bump
If plugin works on NET8 without issue there should be no need to force it up to NET8

**Reply by anna-is-cute** [reply]
Thanks

**Comment by reiichi001** [comment]
New plugin updates should target NET8, but if the existing plugin works, there's no need to make a commit bumping it just for targeting purposes.

A handful of plugins did break during NET8 testing (we had asked users to switch to the net8 branch and let us know), but some will be fixed when plogon targets NET8 and there are one or two that need to actually target NET8 themselves to fix issues.

**Reply by wolfcomp** [reply]
new plugins will be required to target NET8 as they can't be built otherwise when they target `Dalamud.dll`

**Comment by goaaats** [comment]
We'll make it possible for Plogon to build net8.0-windows targets soon-ish(probably today), and the intended workflow is that you can choose either net7.0-windows or net8.0-windows until API 10(expansion release).

I'll try to clarify the news post with more info when we're done.

FWIW, this is a good spot to ask these things and I probably prefer it over discord.
